push/src/detail/label_encoder.h

```diff
@@ -7,7 +7,15 @@
 namespace prometheus {
 namespace detail {
 
-void encodeLabel(std::ostream& os, const Label& label);
+#if defined _WIN32
+  #ifdef __GNUC__
+    void encodeLabel(std::ostream& os, const Label& label) __attribute__ ((dllexport))
+  #else
+    void encodeLabel(std::ostream& os, const Label& label) __declspec(dllexport)
+  #endif
```

Comment:
i was trying to address multi-platform, this should work but i cannot verify this on my local environment.